zerver/lib/stream_subscription.py

```diff
@@ -313,3 +315,94 @@ def get_subscriptions_for_send_message(
         )
     )
     return query
+
+
+def update_user_streams_subscriber_count(activate: bool, user_profile: UserProfile) -> None:
```

Comment:
I think it might be most readable to just have two small functions, `update_subscriber_counts_for_new_user(user_profile)` and `update_subscriber_counts_for_deactivated_user(user_profile)`

I'm also not sure why you filter on `is_user_active` -- that deserves a comment.

Reply:
> I think it might be most readable to just have two small functions, `update_subscriber_counts_for_new_user(user_profile)` and `update_subscriber_counts_for_deactivated_user(user_profile)`

For creating a new user, there is `update_streams_subscriber_count` which is called in `bulk_add_subscriptions`, this leaves `update_user_streams_subscriber_count` more suitable for reactivating/deactivating a user.



> I'm also not sure why you filter on `is_user_active` -- that deserves a comment.

While it's not wrong, yes it's not necessary since it's already guaranteed that all user stream subscriptions will be either with `is_user_active=True` or `False`, so no need to filer by that field.

Reply:
Yeah, I think it's probably clearer to not filter given that we already know that state, one way or another, in the caller.